internal/bminventory/inventory.go

```diff
@@ -5157,6 +5145,26 @@ func (b *bareMetalInventory) getBoundCluster(db *gorm.DB, infraEnv *common.Infra
 	return nil, nil
 }
 
+// Checks whether the specified obj can be updated by the user.
+// If not allowed, checks whether it can be read by the user.
+// This is required in order to return an appropriate error for
+// users who are only allowed to read.
+func (b *bareMetalInventory) checkAccessToObj(ctx context.Context, obj interface{}, objType string, objId *strfmt.UUID) error {
```

Comment:
Maybe make the name a bit more accurate?

```suggestion
func (b *bareMetalInventory) checkUpdateAccessToObj(ctx context.Context, obj interface{}, objType string, objId *strfmt.UUID) error {
```